### PR TITLE
Feature: Rework Properties

### PR DESCRIPTION
**Pull request by dislogical**
Properties now require fields to be provided, and allow for a future of less `std.signals`. This (more importantly) allows for documentation to be generated properly.

### CODE REVIEW

**Comment by wozniakty** [comment]
Looks good.  :+1:

**Comment by tmr9209** [comment]
Looks good, but almost everyones branches will need updates for their respective properties.

File: source/core/gameobject.d [hunk 1]
```diff
@@ -11,37 +11,45 @@ import std.signals, std.conv, std.variant;
...
 	/**
 	 * The current transform of the object.
 	 */
-	mixin Property!( "Transform", "transform", "public" );
+	mixin( Property!_transform );
 	/**
 	 * The Material belonging to the object
 	 */
-	mixin Property!( "Material", "material", "public" );
+	mixin( Property!_material );
```
Comment (wozniakty) [inline source/core/gameobject.d]: So these variables were previously public setters, and no longer are.  Is this a bug in Property, or is this just missing an additional parameter?

**Comment by brooklynlittell** [comment]
Looks great! :+1:

**Pull request merged**